src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.conversion;
+
+import com.microsoft.spring.data.gremlin.common.Constants;
+import com.microsoft.spring.data.gremlin.exception.UnexpectedGremlinSourceTypeException;
+import lombok.NoArgsConstructor;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@NoArgsConstructor
+public class GremlinScriptVertexFindLiteral implements GremlinScript<String> {
+
+    @Override
+    public String generateScript(@NonNull GremlinSource source) {
+        if (!(source instanceof GremlinSourceVertex)) {
+            throw new UnexpectedGremlinSourceTypeException("should be the instance of GremlinSourceVertex");
+        }
+
+        final List<String> scriptList = new ArrayList<>();
+        final String id = source.getId();
+
+        Assert.notNull(id, "id should not be null");
+
+        scriptList.add(Constants.GREMLIN_PRIMITIVE_GRAPH);
+        scriptList.add(String.format(Constants.GREMLIN_PRIMITIVE_VERTEX, id));
```

Comment:
so only can find by id?

Reply:
As the client is binding with one collection, id is unique in a graph.

Reply:
Sorry for misunderstand, current we only implement findById, more will be added later, as signature may change then.